tests/model_serving/model_server/private_endpoint/conftest.py

```diff
@@ -0,0 +1,207 @@
+            },
+        },
+    }
+
+    with InferenceService(
+        client=admin_client, namespace=endpoint_namespace.name, predictor=predictor, name="test"
+    ) as isvc:
+        isvc.wait_for_condition(condition="Ready", status="True")
+        yield isvc
+
+
+@pytest.fixture()
+def storage_config_secret(
+    admin_client: DynamicClient,
+    endpoint_namespace: Namespace,
+    endpoint_s3_secret: Secret,
+    aws_access_key: str,
+    aws_secret_access_key: str,
+) -> Generator[Secret]:
+    secret = {
+        "access_key_id": aws_access_key,
+        "bucket": AWS_BUCKET,
+        "default_bucket": AWS_BUCKET,
+        "endpoint_url": AWS_ENDPOINT,
+        "region": AWS_REGION,
+        "secret_access_key": aws_secret_access_key,
+        "type": "s3",
+    }
+    data = {"endpoint-s3-secret": b64_encoded_string(json.dumps(secret))}
+    with Secret(
+        client=admin_client,
+        namespace=endpoint_namespace.name,
+        data_dict=data,
+        wait_for_resource=True,
+        name="storage-config",
+    ) as storage_config:
+        yield storage_config
+
+
+@pytest.fixture()
+def service_mesh_member(admin_client: DynamicClient, diff_namespace: Namespace) -> Generator[ServiceMeshMember]:
+    with ServiceMeshMember(
+        client=admin_client,
+        namespace=diff_namespace.name,
+        name="default",
+        control_plane_ref={"name": "data-science-smcp", "namespace": "istio-system"},
+        wait_for_resource=True,
+    ) as smm:
+        yield smm
+
+
+@pytest.fixture()
+def endpoint_pod_with_istio_sidecar(admin_client: DynamicClient, endpoint_namespace: Namespace) -> Generator[Pod]:
+    pod = create_sidecar_pod(
+        admin_client=admin_client,
+        namespace=endpoint_namespace.name,
+        istio=True,
+        pod_name="test-with-istio",
+    )
+    yield pod
+    pod.clean_up()
+
+
+@pytest.fixture()
+def endpoint_pod_without_istio_sidecar(admin_client: DynamicClient, endpoint_namespace: Namespace) -> Generator[Pod]:
+    pod = create_sidecar_pod(
+        admin_client=admin_client,
+        namespace=endpoint_namespace.name,
+        istio=False,
+        pod_name="test",
+    )
+    yield pod
+    pod.clean_up()
+
+
+@pytest.fixture()
+def diff_pod_with_istio_sidecar(admin_client: DynamicClient, diff_namespace: Namespace) -> Generator[Pod]:
+    pod = create_sidecar_pod(
+        admin_client=admin_client,
+        namespace=diff_namespace.name,
+        istio=True,
+        pod_name="test-with-istio",
+    )
+    yield pod
+    pod.clean_up()
+
+
+@pytest.fixture()
+def diff_pod_without_istio_sidecar(admin_client: DynamicClient, diff_namespace: Namespace) -> Generator[Pod]:
+    pod = create_sidecar_pod(
+        admin_client=admin_client,
+        namespace=diff_namespace.name,
+        istio=False,
+        pod_name="test",
+    )
+    yield pod
+    pod.clean_up()
+
+
+@pytest.fixture()
+def running_flan_pod(admin_client: DynamicClient, endpoint_isvc: InferenceService) -> None:
```

Comment:
get the deployment and use `wait_for_replicas`

Reply:
why? I have a single pod and need to make sure that it's running